Workflows/Environment Synchronization.md

```diff
@@ -0,0 +1,126 @@
+## Environment Synchronization
+A synchronization of environments, or more precisely, a synchronization of the data used in different environments,
+is usually necessary when you need to work in an environment that is as production-like as possible, but you don't have
+access to production itself, for example:
+* when you need to debug a more complex issue occuring in production
+* when you need data that is as close to production as possible (both when it comes to type and amount)
+for development and/or testing purposes
+
+Access to production is normally limited in order to protect the private information of users, among other things.
+The same should be ensured once this data is copied to another environment. In order to limit the number
+of developers who have access to raw production data, it would be best if that data was anonymized by someone
+from the DevOps team before it is given to the developer. To achieve this, the Rails team has agreed with
+the DevOps team that the synchronization process should look like this:
+1. a completely new environment is set up
+2. the production database is copied to this new environment
+3. anonymization is run on the copied production database
+4. the anonymized database is dumped
+5. the anonymized database dump is copied to a destination of choice (e.g. S3 bucket)
+6. the anonymized database dump is downloaded to the destination environment (staging, development, etc.)
```

Comment:
We should probably add a step for removing the dump (even if its anonymized)  after its pulled by the dev. I hope its not put into a public bucket?

Reply:
The whole AWS setup is something to be agreed with the devops team. They will have to set up a user with the appropriate permissions, etc., but I will add a warning here just so this does not go unsaid.